bottlecap/src/metrics/datadog.rs

```diff
@@ -30,37 +28,24 @@ pub enum ShipError {
     Json(#[from] serde_json::Error),
 }
 
-impl Default for DdApi {
-    fn default() -> Self {
-        Self {
-            api_key: env::var("DD_API_KEY").expect("unable to read envvars, catastrophic failure"),
+impl DdApi {
+    pub fn new(api_key: String) -> Self {
+        DdApi {
+            api_key,
             ureq_agent: ureq::AgentBuilder::new().build(),
         }
     }
-}
-
-impl DdApi {
-    /// Create a new `DdApi` instance
```

Comment:
Makes sense to delete this. By convention, in Rust we use default() instead of new for empty constructors.

Reply:
Thanks – I didn't know that!

Reply:
ah my bad, good to know